testament/categories.nim

```diff
@@ -128,6 +128,9 @@ proc runBasicDLLTest(c, r: var TResults, cat: Category, options: string) =
   var test3 = makeTest("lib/nimhcr.nim", options & " --outdir:tests/dll" & rpath, cat)
   test3.spec.action = actionCompile
   testSpec c, test3
+  var test4 = makeTest("tests/dll/visibility.nim", options & " --app:lib" & rpath, cat)
```

Comment:
could this be a regular test (ie not requiring a custom entry in testament) eg by supporting `$rpath` to be defined as above; eg:
(untested):
```nim
discard """
  cmd: ""nim c --app:lib $rpath $file"
  output: "could not import: foo"
  exitcode: 1
"""
```

Reply:
I need to build a dll + an executable out of it, so it has to be a custom run.

Reply:
here's a self contained example that doesn't require testament to be aware of this test file:
```nim
when defined(timn_D20200119T151729):
  proc fun1() {.exportc, dynlib.} = discard
  proc fun2() {.exportc.} = discard
else:
  import std/[strformat,os,strutils]
  const libF = "/tmp/" / (DynlibFormat % "D20200119T151729")
  const nim = getCurrentCompilerExe()
  const file = currentSourcePath()
  static:
    let cmd = fmt"{nim} c -d:timn_D20200119T151729 --app:lib --passc:-fvisibility=hidden -o:{libF} {file}"
    # echo cmd
    let (output, exitCode) = gorgeEx cmd
    doAssert exitCode == 0, output
  proc fun1() {.importc, dynlib: libF.}
  proc fun2() {.importc, dynlib: libF.}
  fun1()
  fun2() # expected to fail with: could not import: fun2
```

Reply:
So a simple, easy to understand 4 lines change in testament + a self-explanatory test sample vs a complicated piece of code that only works on *nix?

I'm pretty sure I know what I'd choose.